src/experiment_prototype/experiment_slice.py

```diff
@@ -609,14 +619,74 @@ def check_clrfrqrange(cls, clrfrqrange, values):
 
         return values
 
-    @validator('freq')
-    def check_freq(cls, freq):
-        if not freq:
-            return
 
+    @validator('txctrfreq', always=True)
+    def check_txctrfreq(cls, txctrfreq):
+        if not txctrfreq:
+            txctrfreq = 12000.0  # Default value when not set
+
+        # Note - txctrfreq set here and modify the actual center frequency to a
+        # multiple of the clock divider that is possible by the USRP - this default value set
+        # here is not exact (center freq is never exactly 12 MHz).
+
+        # convert from kHz to Hz to get correct clock divider. Return the result back in kHz.
+        clock_multiples = options.usrp_master_clock_rate / 2 ** 32
+        clock_divider = math.ceil(txctrfreq * 1e3 / clock_multiples)
+        txctrfreq = (clock_divider * clock_multiples) / 1e3
+
+        return txctrfreq
+
+    @validator('rxctrfreq', always=True)
+    def check_rxctrfreq(cls, rxctrfreq):
+        if not rxctrfreq:
+            rxctrfreq = 12000.0  # Default value when not set
+
+        # Note - rxctrfreq set here and modify the actual center frequency to a
+        # multiple of the clock divider that is possible by the USRP - this default value set
+        # here is not exact (center freq is never exactly 12 MHz).
+
+        # convert from kHz to Hz to get correct clock divider. Return the result back in kHz.
+        clock_multiples = options.usrp_master_clock_rate / 2 ** 32
+        clock_divider = math.ceil(rxctrfreq * 1e3 / clock_multiples)
+        rxctrfreq = (clock_divider * clock_multiples) / 1e3
+
+        return rxctrfreq
+
+    @validator('freq')
+    def check_freq(cls, freq, values):
```

Comment:
I like the idea, but I think this might be a little bit outside the scope for this particular PR. So I'll make an issue and reference the PR for a potential enhancement to be made in the future.